src/Cryptol/Backend/Concrete.hs

```diff
@@ -340,11 +342,10 @@ instance Backend Concrete where
   -- NB: under the precondition that `m` is prime,
   -- the only values for which no inverse exists are
   -- congruent to 0 modulo m.
-  znRecip sym m x
-    | r == 0    = raiseError sym DivideByZero
-    | otherwise = pure r
-   where
-     r = Integer.recipModInteger x m
+  znRecip sym m x =
```

Comment:
I think it's probably better to make `recipModInteger` return a `Maybe Integer` instead of pushing the unboxed tuples here to the use sites.  I imagine inlining will basically get us back here, but I don't think it's performance-sensitive enough to worry about either.

Reply:
I can certainly do that in the case of `integerRecipMod`. I wouldn't be able to apply exactly the same treatment for `bigNatSub`, as its return type is `(# (# #) | BigNat# #)` and `Maybe BigNat#` won't kind-check on GHC 9.0+. I could concoct an unlifted version of `Maybe` if you'd prefer to avoid any use of `UnboxedSums` at use sites, however.

Reply:
I think it's OK to push unboxed tuples into the `PrimeEC` module, which is doing some pretty low-level hacking itself, I'd just prefer to avoid them in other, less grungy modules.

I think `PrimEC` is the only use site for `bigNatSub`, so that should be fine to leave as-is.